community/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/MergeNodeAcceptanceTest.scala

```diff
@@ -30,4 +30,10 @@ class MergeNodeAcceptanceTest extends ExecutionEngineFunSuite with QueryStatisti
       assertStats(result, nodesCreated = 1, propertiesWritten = 1, labelsAdded = 1)
     }
   }
+
+  test("Merging with self loop and relationship uniqueness") {
+    graph.execute("CREATE (a) CREATE (a)-[:X]->(a)")
+    val result = updateWithBothPlannersAndCompatibilityMode("MERGE (a)-[:X]->(b)-[:X]->(c) RETURN 42")
+    assertStats(result, relationshipsCreated = 2, nodesCreated = 3)
+  }
```

Comment:
Since the fix seems relevant to all predicates, perhaps we should have more tests:

* Test for relationship uniqueness without self-relationships
* Test for other predicates than relationship uniqueness

Reply:
I've notice that the second case is already covered by the fact that Cypher does not allow explicit WHERE clauses for MERGE, and the implicit WHERE from in-lined Label and Property checks was already checked for separately, so the only predicate not covered was relationship uniqueness, which is good. I the meantime I have written a test for the first case.